src/pipdeptree/_models/package.py

```diff
@@ -3,33 +3,36 @@
 import re
 from abc import ABC, abstractmethod
 from importlib import import_module
-from importlib.metadata import PackageNotFoundError, metadata, version
+from importlib.metadata import Distribution, PackageNotFoundError, metadata, version
 from inspect import ismodule
-from typing import TYPE_CHECKING
+from pathlib import Path
+from typing import TYPE_CHECKING, Any
 
-from pip._vendor.pkg_resources import Requirement  # noqa: PLC2701
+from packaging.requirements import Requirement
+from packaging.specifiers import SpecifierSet
+from pip._internal.models.direct_url import DirectUrl  # noqa: PLC2701
+from pip._internal.utils.egg_link import egg_link_path_from_sys_path  # noqa: PLC2701
 
 if TYPE_CHECKING:
-    from pip._internal.metadata import BaseDistribution
-    from pip._vendor.pkg_resources import DistInfoDistribution
+    from importlib.metadata import Distribution
 
 
 def pep503_normalize(name: str) -> str:
-    return re.sub("[-_.]+", "-", name)
+    return re.sub("[-_.]+", "-", name).lower()
+
+
+def contains_extra(marker: str) -> bool:
+    return re.search(r"\bextra\s*==", marker)
```

Comment:
Since, according to the [docs](https://docs.python.org/3/library/re.html#re.search), `re.search()` can return `None`, let's change this line to the following:
```suggestion
    return re.search(r"\bextra\s*==", marker) is not None
```

Reply:
Done